tests/hikari/impl/test_rest.py

```diff
@@ -292,6 +292,199 @@ def test_invalidate_when_token_is_stored_token(self):
+        class MockLock:
+            def __init__(self, strat):
+                self._strategy = strat
+
+            async def __aenter__(self):
+                self._strategy._token = "cab.cab.cab"
+                return self
+
+            async def __aexit__(self, exc_type, exc_val, exc_tb):
+                return
+
+        mock_rest = mock.AsyncMock()
+        strategy = rest.OAuthCredentialsStrategy(client=123456789, client_secret="123123123", auth_code="auth#code", redirect_uri="https://web.site/auth/discord")
+        strategy._lock = MockLock(strategy)
+        strategy._token = None
+        strategy._expire_at = time.monotonic() + 500
+
+        result = await strategy.acquire(mock_rest)
+
+        assert result == "cab.cab.cab"
+
+        mock_rest.authorize_access_token.assert_not_called()
+
+    @pytest.mark.asyncio()
+    async def test_acquire_caches_client_http_response_error(self):
+        mock_rest = mock.AsyncMock()
+        error = errors.ClientHTTPResponseError(
+            url="okokok", status=42, headers={}, raw_body=b"ok", message="OK", code=34123
+        )
+        mock_rest.authorize_access_token.side_effect = error
+        strategy = rest.OAuthCredentialsStrategy(client=123456789, client_secret="123123123", auth_code="auth#code", redirect_uri="https://web.site/auth/discord")
+
+        with pytest.raises(errors.ClientHTTPResponseError):
+            await strategy.acquire(mock_rest)
+
+        with pytest.raises(errors.ClientHTTPResponseError):
+            await strategy.acquire(mock_rest)
+
+        mock_rest.authorize_access_token.assert_awaited_once_with(
+                        client=123456789,
+                        client_secret="123123123",
+                        code="auth#code",
+                        redirect_uri="https://web.site/auth/discord",
+                    )
+
+    def test_invalidate_when_token_is_not_stored_token(self):
+        strategy = rest.OAuthCredentialsStrategy(client=123456789, client_secret="123123123", auth_code="auth#code", redirect_uri="https://web.site/auth/discord")
+        strategy._expire_at = 10.0
+        strategy._token = "token"
+
+        strategy.invalidate("tokena")
+
+        assert strategy._expire_at == 10.0
+        assert strategy._token == "token"
+
+    def test_invalidate_when_no_token_specified(self):
+        strategy = rest.OAuthCredentialsStrategy(client=123456789, client_secret="123123123", auth_code="auth#code", redirect_uri="https://web.site/auth/discord")
+        strategy._expire_at = 10.0
+        strategy._token = "token"
+
+        strategy.invalidate(None)
+
+        assert strategy._expire_at == 0.0
+        assert strategy._token is None
+
+    def test_invalidate_when_token_is_stored_token(self):
+        strategy = rest.OAuthCredentialsStrategy(client=123456789, client_secret="123123123", auth_code="auth#code", redirect_uri="https://web.site/auth/discord")
```

Comment:
This strategy could be extracted to a fixture and used across all tests, instead of having to instantiate it everywhere